UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/AuthTokenProvider.cs

```diff
@@ -0,0 +1,126 @@
+using static UKHO.ExchangeSetService.API.FunctionalTests.Helper.TestConfiguration;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.Helper
+{
+    public class AuthTokenProvider
+    {
+        static string EssAccessToken = null;
+        static string EssAccessTokenNoAuth = null;
+        static EssAuthorizationTokenConfiguration EssauthConfig = new TestConfiguration().EssAuthorizationConfig;
+
+        public async Task<string> GetEssToken()
+        {
+            EssAccessToken = await GenerateEssToken(EssauthConfig.AutoTestClientId, EssauthConfig.AutoTestClientSecret, EssAccessToken);
+            return EssAccessToken;
+        }
+
+        public async Task<string> GetEssTokenNoAuth()
+        {
+            EssAccessTokenNoAuth = await GenerateEssToken(EssauthConfig.AutoTestClientIdNoAuth, EssauthConfig.AutoTestClientSecretNoAuth, EssAccessTokenNoAuth);
+            return EssAccessTokenNoAuth;
+        }
+
+        /// <summary>
+        /// GenerateToken
+        /// </summary>
+        /// <param name="ClientId"></param>
+        /// <param name="ClientSecret"></param>
+        /// <param name="Token"></param>
+        /// <returns></returns>
+
+        private static async Task<string> GenerateEssToken(string ClientId, string ClientSecret, string Token)
+        {
+
+            string[] scopes = new string[] { $"{EssauthConfig.EssClientId}/.default" };
+            if (EssauthConfig.IsRunningOnLocalMachine)
+            {
+                if (Token == null)
+                {
+                    IPublicClientApplication debugapp = PublicClientApplicationBuilder.Create(EssauthConfig.EssClientId).
+                                                        WithRedirectUri("http://localhost").Build();
+
+
+
+                    //Acquiring token through user interaction
+                    AuthenticationResult tokenTask = await debugapp.AcquireTokenInteractive(scopes)
+                                                            .WithAuthority($"{EssauthConfig.MicrosoftOnlineLoginUrl}{EssauthConfig.TenantId}", true)
+                                                            .ExecuteAsync();
+                    Token = tokenTask.AccessToken;
+                }
+                return Token;
+            }
+            else
+            {
+                if (Token == null)
+                {
+                    IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(ClientId)
+                                                    .WithClientSecret(ClientSecret)
+                                                    .WithAuthority(new Uri($"{EssauthConfig.MicrosoftOnlineLoginUrl}{EssauthConfig.TenantId}"))
+                                                    .Build();
+
+                    AuthenticationResult tokenTask = await app.AcquireTokenForClient(scopes).ExecuteAsync();
+                    Token = tokenTask.AccessToken;
+                }
+                return Token;
+            }
+        }
```

Comment:
Null check on token is repetitive. It can be simplified as:
```suggestion
       private static async Task<string> GenerateEssToken(string ClientId, string ClientSecret, string Token)
        {
            string[] scopes = new string[] { $"{EssauthConfig.EssClientId}/.default" };
            if (Token == null) 
            {
                if (EssauthConfig.IsRunningOnLocalMachine)
                {
                    IPublicClientApplication debugApp = PublicClientApplicationBuilder.Create(EssauthConfig.EssClientId).
                                                        WithRedirectUri("http://localhost").Build();

                    //Acquiring token through user interaction
                    AuthenticationResult tokenTask = await debugApp.AcquireTokenInteractive(scopes)
                                                            .WithAuthority($"{EssauthConfig.MicrosoftOnlineLoginUrl}{EssauthConfig.TenantId}", true)
                                                            .ExecuteAsync();
                    Token = tokenTask.AccessToken;
                }
		else
		{
			IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(ClientId)
                                                    .WithClientSecret(ClientSecret)
                                                    .WithAuthority(new Uri($"{EssauthConfig.MicrosoftOnlineLoginUrl}{EssauthConfig.TenantId}"))
                                                    .Build();

                    	AuthenticationResult tokenTask = await app.AcquireTokenForClient(scopes).ExecuteAsync();
                    	Token = tokenTask.AccessToken;
		}
                
            }
	    return Token;
          }
```